components/user/language/ChangeLanguage.tsx

```diff
@@ -27,12 +27,29 @@ const ChangeLanguage = () => {
   const [modalVisible, setModalVisible] = useState(false);
   const [selectedLanguage, setSelectedLanguage] = useState('English');
   const [selectedVisibleLanguage, setSelectedVisibleLanguage] = useState('ENG');
-  const [selectedVisibleLanguageFlag, setSelectedVisibleLanguageFlag] = useState(require('../../../assets/lang/eng.png'));
+  const [selectedVisibleLanguageFlag, setSelectedVisibleLanguageFlag] = useState(require('../../../assets/lang/emp.png'));
   const { add, remove } = useNotifications();
 
   const [mutateUserLanguage, { isError: isUserLanguageError, error: userLanguageError, isLoading: isUserLanguageLoading }] =
     useSetUserLanguageMutation();
 
+  const {
+    data: userLanguageData,
+    isFetching: isUserLanguageDataFetching,
+    refetch: userLanguageRefetch,
+  } = useGetUserLanguageQuery();
+
+  useEffect(() => {
+    if (userLanguageData) {
+      const userLang = languages.find(language => language.code === userLanguageData.language.toUpperCase());
+      if (userLang) {
```

Comment:
move all of this out of useEffect, handle initially undefined data while rendering

Reply:
```
  const {
    data: userLanguageData,
    isFetching: isUserLanguageDataFetching,
    refetch: userLanguageRefetch,
  } = useGetUserLanguageQuery();

  const userLang = languages.find(language => language.code === userLanguageData?.language.toUpperCase());
  setSelectedLanguage(userLang?.name);
  setSelectedVisibleLanguage(userLang?.code);
  setSelectedVisibleLanguageFlag(userLang?.flag);
  ```

Reply:
```
  const [selectedVisibleLanguage, setSelectedVisibleLanguage] = useState<string | undefined>('ENG');
  const [selectedVisibleLanguageFlag, setSelectedVisibleLanguageFlag] = useState<string | undefined>(require('../../../assets/lang/emp.png'));
```